### DISCUSSION TITLE
liquidsoap - segmentation fault

### DISCUSSION THREAD

**Discussion by suwofis** [savonet/liquidsoap#4107] [User Support]
Hello,

I installed the liquidsoap software on a Raspberry Pi 4B. Unfortunately, the error mentioned above occurs every time, even though I do everything according to the instructions. The crazy thing: It runs on an Ubuntu computer - but it has to be on the Raspberry because my NAS with the media data is connected there.

Can anyone explain this strange behavior and how to remedy it?

Thank you very much and best regards.

**Comment by toots** [comment]
This is a known issue due to some issues with the rasbian ffmpeg packages. There's a whole thread here: https://github.com/savonet/liquidsoap/issues/3491

The last comment has a solution:

> For anyone coming here looking for a solution, here's a workaround I came up with that combines and tweaks the suggestions above. It's not a good permanent solution, but it should work for now.
> 
> This process will work even if you've already installed liquidsoap and are getting the segfault.
> 
> Tested on a fresh install of RaspberryOS 12.5 with liquidsoap 2.1.3-2.
> 
> Create the /etc/apt/preferences.d/ffmpeg.pref file with the following contents:
> 
> ```
> Package: ffmpeg libavcodec-dev libavcodec59 libavdevice59 libavfilter8 libavformat-dev libavformat59 libavutil-dev libavutil57 libpostproc56 libswresample-dev libswresample4 libswscale-dev libswscale6
> Pin: origin deb.debian.org
> Pin-Priority: 1001
> ```
> (This tells apt to only get these packages from the Debian repos, not the archive.raspberrypi.com repos. The priority of 1001 tells apt that we want it to downgrade packages, if needed.)
> 
> Run apt to downgrade (or install) the problematic packages:
> 
> ```
> apt install ffmpeg libswresample4 libavdevice59 libavfilter8 libpostproc56 libavutil57 libswscale6 libavformat59 libavcodec59
> ```
>
> That's it. liquidsoap should run without segfaulting and you can run updates as normal.

**Reply by suwofis** [reply]
Hey, thanks. That worked. Now I have the problem that no playlist can be read. Maybe I make a new thread because it doesn't belong to that topic. But thanks again :)
